homeassistant/components/input_number/__init__.py

```diff
@@ -306,6 +306,7 @@ async def async_update_config(self, config: dict) -> None:
         """Handle when the config is updated."""
         self._config = config
         # just in case min/max values changed
-        self._current_value = min(self._current_value, self._maximum)
-        self._current_value = max(self._current_value, self._minimum)
-        self.async_write_ha_state()
+        if self._current_value is not None:
+            self._current_value = min(self._current_value, self._maximum)
+            self._current_value = max(self._current_value, self._minimum)
+            self.async_write_ha_state()
```

Comment:
min/max required non-null values so adding a check here.
I don't think we need `async_write_ha_state` if we don't change the value.